sys/net/link_layer/ieee802154/ieee802154.c

```diff
@@ -27,6 +27,14 @@ size_t ieee802154_set_frame_hdr(uint8_t *buf, const uint8_t *src, size_t src_len
     uint8_t type = (flags & IEEE802154_FCF_TYPE_MASK);
     uint8_t bcast = (flags & IEEE802154_BCAST);
 
+    if (dst_pan.u16 == src_pan.u16 &&
+        (src_len != 0 && dst_len != 0)) {
```

Comment:
@miri64 I don't understand why src address and length are set in `_send` [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_ieee802154.c#L172) but the destination is not set the same way when it is multicast? I mean the code in `gnrc_netdev2_ieee802154.c` already _knows_ about L2 so it should be able to set the address like [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/link_layer/ieee802154/ieee802154.c#L60), or not?